### ISSUE TITLE
classIntervals doesn't like POSIXct

### ISSUE THREAD

**Issue by edzer** [r-spatial/classInt#8]
```r
> classInt::classIntervals(Sys.time()+1:10)
Error in classInt::classIntervals(Sys.time() + 1:10) : var is not numeric
> classInt::classIntervals(as.numeric(Sys.time()+1:10))
style: quantile
  one of 126 possible partitions of this variable into 5 classes
[1563184092,1563184094) [1563184094,1563184096) [1563184096,1563184098) 
                      2                       2                       2 
[1563184098,1563184100) [1563184100,1563184101] 
                      2                       2 
> quantile(Sys.time()+1:10, c(0,.25,.5,.75,1))
                        0%                        25% 
"2019-07-15 11:50:12 CEST" "2019-07-15 11:50:15 CEST" 
                       50%                        75% 
"2019-07-15 11:50:17 CEST" "2019-07-15 11:50:19 CEST" 
                      100% 
"2019-07-15 11:50:21 CEST" 
```
I'm now working around this in `stars` and `sf`, but I don't see the point why this shouldn't work.

**Comment by rsbivand** [comment]
The `stats:::quantile.POSIXt()` method does:
```
quantile.POSIXt <- function(x, ...)
    .POSIXct(quantile(unclass(as.POSIXct(x)), ...), attr(x, "tzone"))
```
Is this what is needed? For which styles?

**Comment by edzer** [comment]
Ideally for all; what you'd need to do is an `as.numeric` on entry, and set the class back just before return. Thinking about it, it would be good if it worked for `units` as well. You could do this by stripping all attributes on entry, and setting them all back just before return.

**Comment by rsbivand** [comment]
Could you give me a couple of examples matching your needs? I need to adapt the other functions in the package.

**Comment by edzer** [comment]
```r
library(classInt)

x = Sys.time() + 1:10
try(classIntervals(x))
# Error in classIntervals(x) : var is not numeric

library(units)
# udunits system database from /usr/share/xml/udunits
x = set_units(1:15, km/h)
(i = classIntervals(x))
# style: quantile
#   one of 1,001 possible partitions of this variable into 5 classes
#    [1 [km/h],3.8 [km/h])  [3.8 [km/h],6.6 [km/h])  [6.6 [km/h],9.4 [km/h]) 
#                        3                        3                        3 
# [9.4 [km/h],12.2 [km/h])  [12.2 [km/h],15 [km/h]] 
#                        3                        3 
i$brks
# Units: [km/h]
# [1]  1.0  3.8  6.6  9.4 12.2 15.0
```
(Interestingly, although the `units` case already works, the intervals returned are not those you'd want to see in a figure or map legend, IMO, but then for instance `sf` works directly with the `$brks`)

**Comment by rsbivand** [comment]
Yes, I need the use cases where $brks bites. Would classInt need conditionally to import **units** to add decorations to the legend?

**Comment by edzer** [comment]
In principle yes, but everything seems fine as it is ($var, $brks); the units appearing in the `print` method of `classIntervals` objects do not appear in any legend, do they?

**Comment by rsbivand** [comment]
Does 07005f4ce2dab3a0660bd4d9035fc8ceec5f280c provide what is needed?

**Comment by rsbivand** [comment]
@edzer: could you please add the hooks to install units and its external dependencies, to get around the travis failure?

**Comment by rsbivand** [comment]
@edzer : is this good to go? May I close and submit to CRAN?

**Comment by rsbivand** [comment]
Released 0.4-1 on CRAN.

**Comment by edzer** [comment]
great, thanks!!
